File: php/difference-of-squares/difference-of-squares.php
<?php

function sumOfSquares($number)
{
    $numbers = range(1, $number);
    $fn = function ($x) {
        return pow($x, 2);
    };
    return array_sum(array_map($fn, $numbers));
}

function squareOfSums($number)
{
    $numbers = range(1, $number);
    return pow(array_sum($numbers),2);
}

function difference($number)
{
    return squareOfSums($number) - sumOfSquares($number);
}

File: ruby/difference-of-squares/difference_of_squares.rb
module BookKeeping
  VERSION = 4
end


class Squares
  
  def initialize num
    @range = (1..num).to_a
  end
  
  def self.sqrt
    self * self
  end
  
  def square_of_sum
    @range.reduce(&:+).sqrt
  end
  
  def sum_of_squares
    @range.map(&:sqrt).reduce(&:+)
  end
  
  def difference
    square_of_sum - sum_of_squares
  end
  
end